pkg/controller/crdmirroring/types/types.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+const (
+	ManagedByController = "crd.antrea.io/managed-by-mirroring-controller"
+	ControllerName      = "crdmirroring-controller"
```

Comment:
The key and the value is redundant, we should either use "crd.antrea.io/managed-by: crdmirroring-controller" or just "crd.antrea.io/managed-by-mirroring-controller:". For this case, the former sounds better to me as "managed-by" allows other values in the future. I guess you are trying to unify the two annotations here, but it doesn't look not very necessary.

Reply:
> still don't see this is updated except the method name is updated. This could have been one line method if the DeepCopy and annotation mutation is moved to controller
I thought of moving the DeepCopy and annotation mutation to controller. However,  metav1.Object does't have a DeepCopy method. Only a specific type of CRD has a DeepCopy method.

Reply:
> The key and the value is redundant, we should either use "crd.antrea.io/managed-by: crdmirroring-controller" or just "crd.antrea.io/managed-by-mirroring-controller:". For this case, the former sounds better to me as "managed-by" allows other values in the future. I guess you are trying to unify the two annotations here, but it doesn't look not very necessary. 

Yes, I just want to unify the two annotations. I also modified the code in `pkg/controller/crdmirroring/crdmirroring_handlers.go` as below. I want that the existence of annotations is the only condition, not the value. So does in the code of controller part. The values of annotation can be any string. I don't know if it is necessary to define it in const.

```go
func (c *Controller) onNewCRDAdd(obj interface{}) {
	crd := obj.(metav1.Object)

	_, exist := crd.GetAnnotations()[types.ManagedByController]
	if exist {
		klog.V(4).Infof("Processing mirroring %s %s/%s ADD event", c.crdName, crd.GetNamespace(), crd.GetName())
		c.queueCRD(obj)
	}
}
```
`

Reply:
I think it's not necessary to unify annotations to key-values or only keys, especially when it makes the annotations read unnaturally.
Some annotations are key-value while others are kind of mark (no value is necessary). Of course using "managed-by-mirroring-controller" can work for mirroring case, but "managed-by" itself is very generic and exclusive, we cannot predict whether we will have other cases in which we need to annotate "managed-by" other components, which would lead to two annotations "managed-by-A" and "managed-by-B" will be awkward and unmaintainable (the code need to scan all annotations to replace managed-by). 
So I would suggest to use original annotation "managed-by: crdmirroring-controller" for new CR, and "stop-mirror:" for legacy CR.